Tempo/Views/ExpandedPlayerView.swift

```diff
@@ -0,0 +1,314 @@
+	@IBOutlet weak var likeButton: UIButton!
+    @IBOutlet weak var likeButtonImage: UIImageView!
+    @IBOutlet weak var addButton: UIButton!
+    @IBOutlet weak var collapseButton: UIButton!
+    @IBOutlet weak var addButtonImage: UIImageView!
+	
+	var progressIndicator: UIView!
+	
+	var postsLikable = false
+	var postHasInfo = false
+	var parentNav: PlayerNavigationController?
+	
+	var songStatus: SavedSongStatus = .NotSaved
+	var post: Post?
+	private var wasPlaying = false
+	
+	var tapGestureRecognizer: UITapGestureRecognizer?
+	var panGestureRecognizer: UIPanGestureRecognizer?
+	var initialPanView: UIView?
+	
+	func setup(parent: PlayerNavigationController) {
+		backgroundColor = UIColor.tempoSuperDarkGray
+		// Setup gesture recognizers
+		tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(ExpandedPlayerView.expandedCellTapped(_:)))
+		tapGestureRecognizer?.delegate = self
+		tapGestureRecognizer?.cancelsTouchesInView = false
+		self.addGestureRecognizer(tapGestureRecognizer!)
+		panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(ExpandedPlayerView.progressPanned(_:)))
+		panGestureRecognizer?.delegate = self
+		panGestureRecognizer?.delaysTouchesBegan = false
+		self.addGestureRecognizer(panGestureRecognizer!)
+		
+		let closeTapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(ExpandedPlayerView.collapseTap(_:)))
+		closeTapGestureRecognizer.cancelsTouchesInView = false
+		topViewContainer.addGestureRecognizer(closeTapGestureRecognizer)
+		let cellPanGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(ExpandedPlayerView.collapsePan(_:)))
+		cellPanGestureRecognizer.delaysTouchesBegan = false
+		topViewContainer.addGestureRecognizer(cellPanGestureRecognizer)
+		
+		parentNav = parent
+		progressView.playerDelegate = parentNav
+		progressView.backgroundColor = UIColor.tempoSuperDarkRed
+		
+		progressIndicator = UIView(frame: CGRectMake(progressView.frame.origin.x - 6, progressView.frame.origin.y - 6, 12, 12))
+		progressIndicator.layer.cornerRadius = 6
+		progressIndicator.backgroundColor = UIColor.tempoLightRed
+		progressIndicator.userInteractionEnabled = true
+		self.addSubview(progressIndicator)
+		self.bringSubviewToFront(playToggleButton)
+		
+		progressView.indicator = progressIndicator
+		
+		playToggleButton.layer.cornerRadius = 5
+		playToggleButton.clipsToBounds = true
+		
+		updateAddButton()
+		likeButton.userInteractionEnabled = false
+		
+		setupMarqueeLabel(songLabel)
+		setupMarqueeLabel(artistLabel)
+	}
+	
+	func updateCellInfo(newPost: Post) {
+		post = newPost
+		songLabel.text = newPost.song.title
+		artistLabel.text = newPost.song.artist
+		albumImage.hnk_setImageFromURL(newPost.song.smallArtworkURL ?? NSURL())
+		if postHasInfo {
+			let time = getPostTime(newPost.relativeDate())
+			postDetailLabel.text = "\(newPost.user.name) posted \(time) ago"
+		} else {
+			postDetailLabel.text = ""
+		}
```

Comment:
lines 92 - 97 can be condensed to 
`postDetailLabel.text = postHasInfo ? "\(newPost.user.name) posted \(getPostTime(newPost.relativeDate())) ago" : ""`